js/components/operatorSignIn/operatorSelection.tsx

```diff
@@ -47,13 +48,15 @@ export const OperatorSelection = ({
         id={inputId}
         inputMode="numeric"
         onChange={(evt) => {
-          setBadgeEntry(evt.target.value);
+          setBadgeEntry({ number: evt.target.value, method: "manual" });
         }}
       />
       <button
         disabled={!buttonEnabled}
         onClick={() => {
-          onOK(badgeEntry);
+          // badgeEntry cannot be null; otherwise the button would be disabled!
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          onOK(badgeEntry!);
```

Comment:
**suggestion:**  it's a bit more verbose, but rather than doing a non-null assertion, which requires disabling an ESLint rule and loses us a bit of type-safety, this is the sort of case where I'd usually prefer to do something like:

```
if (value !== null) {
  doThing(value) // TypeScript knowns it's not null
} else {
  // report an error, maybe by throwing, `console.error`-ing, or sending an exception to Sentry.
}
```

Reply:
Since this is an (in theory) impossible state, I changed it to throw.